addons/purchase_stock/models/stock_rule.py

```diff
@@ -234,7 +234,11 @@ def _prepare_purchase_order_line(self, product_id, product_qty, product_uom, com
             lang=partner.lang,
             partner_id=partner.id,
         )
-        name = product_lang.display_name
+        name_product_description = product_lang.with_context(supplier_info=seller).name_get()
```

Comment:
Why not simply adding supplier_info in the context here ? 
```python
    product_lang = product_id.with_prefetch().with_context(
        lang=partner.lang,
        partner_id=partner.id,
        supplier_info=seller,
    )
```

Reply:
Indeed it works too, I've changed it and I'll ping you once the checks have passed :smile: